bindings/matrix-sdk-ffi/src/identity_status_change.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(uniffi::Record)]
+pub struct IdentityStatusChange {
+    /// The user ID of the user whose identity status changed
+    pub user_id: String,
+
+    /// The new state of the identity of the user. One of:
+    ///
+    /// * Verified
+    /// * Pinned
+    /// * PinViolation
+    /// * PreviouslyVerified
```

Comment:
What about `VerificationViolation` to be consistent with the `has_verification_violation()` method on UserIdentity?

Reply:
Done in 28a8d5468e and 06d97e9230 .

What about `OwnUserIdentityVerifiedState::PreviouslyVerifiedButNoLonger` and `SenderDataType::SenderUnverifiedButPreviouslyVerified` and `VerificationLevel::PreviouslyVerified` - should those be renamed too? (Later, not this PR if so)

Reply:
Right we need to make all that consistent

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/4067